core/src/main/java/com/scalar/db/api/Operation.java

```diff
@@ -9,72 +9,59 @@
 import java.util.Optional;
 import javax.annotation.Nonnull;
 import javax.annotation.concurrent.NotThreadSafe;
-import org.slf4j.Logger;
-import org.slf4j.LoggerFactory;
 
 /**
- * An abstraction for storage operations.
+ * A multidimensional map abstraction for storage operations.
  *
  * @author Hiroyuki Yamada
  */
 @NotThreadSafe
-public abstract class Operation {
-  private static final Logger logger = LoggerFactory.getLogger(Operation.class);
-
+public abstract class Operation extends BaseOperation {
   private final Key partitionKey;
   private final Optional<Key> clusteringKey;
-  private Optional<String> namespace;
-  private Optional<String> tableName;
-  private Consistency consistency;
 
   public Operation(Key partitionKey, Key clusteringKey) {
     this.partitionKey = checkNotNull(partitionKey);
     this.clusteringKey = Optional.ofNullable(clusteringKey);
-    namespace = Optional.empty();
-    tableName = Optional.empty();
-    consistency = Consistency.SEQUENTIAL;
   }
 
   public Operation(Operation operation) {
+    super(operation);
     this.partitionKey = operation.partitionKey;
     this.clusteringKey = operation.clusteringKey;
-    namespace = operation.namespace;
-    tableName = operation.tableName;
-    consistency = operation.consistency;
   }
 
   /**
    * Returns the namespace for this operation
    *
    * @return an {@code Optional} with the returned namespace
    */
+  @Override
   @Nonnull
   public Optional<String> forNamespace() {
-    return namespace;
+    return super.forNamespace();
```

Comment:
I think you can remove these verbose overridden methods as the superclass's methods are called